care/facility/api/viewsets/patient.py

```diff
@@ -147,8 +147,23 @@ def filter_by_category(self, queryset, name, value):
         field_name="last_consultation__symptoms_onset_date"
     )
     last_consultation_admitted_bed_type_list = MultiSelectFilter(
-        field_name="last_consultation__current_bed__bed__bed_type"
+        field_name="last_consultation__current_bed__bed__bed_type",
+        method="filter_by_bed_type",
     )
+
+    def filter_by_bed_type(self, queryset, name, value):
+        values = value.split(",")
+        filter_q = Q()
+
+        for val in values:
+            val = val.strip()
+            if val == "None":
+                filter_q |= Q(last_consultation__current_bed__isnull=True)
+            else:
+                filter_q |= Q(last_consultation__current_bed__bed__bed_type=val)
+
+        return queryset.filter(filter_q)
```

Comment:
you can just query `last_consultation__current_bed__bed__bed_type__in=values` instead of a for loop

Reply:
But we don't have the "None" as value in the choices of `last_consultation__current_bed__bed_type` rgt?? So how does the above query work??

Reply:
https://docs.djangoproject.com/en/4.2/ref/models/querysets/#in
and IN operator is a shorthand for multiple OR operators in SQl so None would be ignored

But you won't get `None` in a list, it'll be just `None` or list of bed types

Reply:
Ohh okay.
I considered the case where "None"  would be coming in a list.(as it's simply one of the IDs of the const object of bed types they have in frontend)
Coz the frontend functionality is a multi select filter. So there might be a case where they will querying for "all the patients who are having an ICU bed alloted or no bed alloted".
If you're sure that we won't be getting the "None" option in a list of options from the frontend then do lemme know. I will make the changes accordingly.

Reply:
No you won't get None within the csv list, but instead the `value` can be an **empty string** or **None Type** in python